### DISCUSSION TITLE
Do we need SQS?

### DISCUSSION THREAD

**Discussion by popovitsj** [aws-samples/websocket-chat-application#43] [General]
Hi,

I noticed in `onconnect` lambda, when a user connects, we use SQS to trigger a lambda that broadcasts to all other users that this user has come online.
In the `onmessage` lambda, we don't use SQS, but broadcast directly.

Do we really need SQS in the `onconnect` handler? Can't we just broadcast from the `onconnect` lambda directly? Is there a trade-off here?

**Comment by santatamas** [comment]
Interesting question! The reason I decided to use the SQS is 2-fold, first it makes the onConnect handler quicker to respond to the connecting user (broadcasting might take a while!), but also - the user status update is not a critical path of the application, hence it can be made asynchronous using the queue. Noone minds if they get the notification with a bit of a lag, but sending messages should be as instant as possible. With that said, that only scales to a certain limit. For broadcasting to a large amount of users, you should still consider using an SQS queue, or use AppSync instead of this solution.

**Reply by popovitsj** [reply]
Thanks, I understand the trade off now.
